Don't compare `pkl` files since they are not always equal byte to byte

Example of failed test:
```bash
FAILED modin/pandas/test/test_io.py::TestPickle::test_to_pickle - AssertionError: assert False
 +  where False = assert_files_eq('/tmp/pytest-of-runner/pytest-6/test_to_pickle0/0539a72a78e111eeb4ad6045bda91651.pkl', '/tmp/pytest-of-runner/pytest-6/test_to_pickle0/0539a85678e111eeb4ad6045bda91651.pkl')
= 1 failed, 1934 passed, 122 skipped, 1272 deselected, 192 xfailed, 1196 warnings in 330.41s (0:05:30) =
```

--- a/modin/pandas/test/test_io.py
+++ b/modin/pandas/test/test_io.py
@@ -3061,14 +3061,14 @@ class TestPickle:
         )
 
     def test_to_pickle(self, tmp_path):
-        modin_df, pandas_df = create_test_dfs(TEST_DATA)
-        eval_to_file(
-            tmp_path,
-            modin_obj=modin_df,
-            pandas_obj=pandas_df,
-            fn="to_pickle",
-            extension="pkl",
-        )
+        modin_df, _ = create_test_dfs(TEST_DATA)
+
+        unique_filename_modin = get_unique_filename(extension="pkl", data_dir=tmp_path)
+
+        modin_df.to_pickle(unique_filename_modin)
+        recreated_modin_df = pd.read_pickle(unique_filename_modin)
+
+        df_equals(modin_df, recreated_modin_df)
 
 
 @pytest.mark.filterwarnings(default_to_pandas_ignore_string)

